WooCommerce/Classes/ViewRelated/Orders/Order Details/Address Edit/EditAddressFormViewModel.swift

```diff
@@ -0,0 +1,74 @@
+import Yosemite
+
+final class EditAddressFormViewModel: ObservableObject {
+
+    init(address: Address?) {
+        self.originalAddress = address
+        updateFieldsWithOriginalAddress()
+    }
+
+    /// Original `Address` model.
+    ///
+    private let originalAddress: Address?
+
+    // MARK: User Fields
+
+    @Published var firstName: String = ""
+    @Published var lastName: String = ""
+    @Published var email: String = ""
+    @Published var phone: String = ""
+
+    // MARK: Address Fields
+
+    @Published var company: String = ""
+    @Published var address1: String = ""
+    @Published var address2: String = ""
+    @Published var city: String = ""
+    @Published var postcode: String = ""
+
+    // MARK: Navigation and utility
+
+    /// Set it to `true` to present the country selector.
+    ///
+    @Published var showCountrySelector: Bool = false
+
+    /// Return `true` if the done button should be enabled.
+    ///
+    var isDoneButtonEnabled: Bool {
+        guard let originalAddress = originalAddress else {
+            return true
+        }
```

Comment:
If we pass nil as the `Original Address` then the button will be enabled by default, is that expected?

Reply:
Fixed by updated comparison flow + non-nil `originalAddress`.
Added tests in 0e3f8d79fe9072ad29dc9f8776fffb55ae6e5025.